File: dice_poker/combination_factory.py
from typing import List
from .combination import (
    Pair, TwoPairs, ThreeOfKind, FourOfKind, FiveOfKind,
    FullHouse, SmallStraight, LargeStraight, Chance, NumberCombination, Combination
)
from .combination_type import CombinationType

def create_all() -> List[Combination]:
    combos: List[Combination] = []
    # Upper section
    combos.append(NumberCombination(1, CombinationType.ONES))
    combos.append(NumberCombination(2, CombinationType.TWOS))
    combos.append(NumberCombination(3, CombinationType.THREES))
    combos.append(NumberCombination(4, CombinationType.FOURS))
    combos.append(NumberCombination(5, CombinationType.FIVES))
    combos.append(NumberCombination(6, CombinationType.SIXES))
    # Lower section
    combos.append(Pair())
    combos.append(TwoPairs())
    combos.append(ThreeOfKind())
    combos.append(FourOfKind())
    combos.append(FiveOfKind())
    combos.append(FullHouse())
    combos.append(SmallStraight())
    combos.append(LargeStraight())
    combos.append(Chance())
    return combos

File: src/dicepoker/combination/CombinationFactory.java
package dicepoker.combination;

import java.util.ArrayList;
import java.util.List;

import dicepoker.model.CombinationType;

public class CombinationFactory {
    public static List<Combination> createAll(){
        List<Combination> list=new ArrayList<>();
        list.add(new Pair()); list.add(new TwoPairs()); list.add(new ThreeOfKind());
        list.add(new FourOfKind()); list.add(new FiveOfKind()); list.add(new FullHouse());
        list.add(new Straight(CombinationType.SMALL_STRAIGHT));
        list.add(new Straight(CombinationType.LARGE_STRAIGHT));
        list.add(new Chance());
        list.add(new NumberCombination(CombinationType.ONES,1));
        list.add(new NumberCombination(CombinationType.TWOS,2));
        list.add(new NumberCombination(CombinationType.THREES,3));
        list.add(new NumberCombination(CombinationType.FOURS,4));
        list.add(new NumberCombination(CombinationType.FIVES,5));
        list.add(new NumberCombination(CombinationType.SIXES,6));
        return list;
    }
}
